### DISCUSSION TITLE
Is it possible to execute command on focus move to a window with specific class/title?

### DISCUSSION THREAD

**Discussion by kimifish** [i3/i3#5912] [Q&A] [answered]
I need quite specific keyboard and power settings in certain applications. Can I automatically run shell script with one argument when moving the focus to a window with a certain class and a title and run it again, with a different arg, with focus loss?

**Comment by orestisfl** [comment, marked as answer]
Yes, check the i3 IPC interface: https://i3wm.org/docs/ipc.html

You can subscribe and react to window focus events
